### ISSUE TITLE
Build fails, protobuf needs to handle russian characters in the path of build command

### ISSUE THREAD

**Issue by MironAtHome** [microsoft/onnxruntime#284]
**Is your feature request related to a problem? Please describe.**
Build fails, protobuf needs to handle russian characters in the path of build command
Command:
C:\<path_in_russian>\microsoft\onnxruntime>build\Windows\Debug\external\protobuf\cmake\Debug\protoc.exe   --cpp_out C:/<path_in_russian>/microsoft/onnxruntime/build/Windows/Debug/onnx -I C:/<path_in_russian>/microsoft/onnxruntime/onnxruntime/core/protobuf C:/<path_in_russian>/microsoft/onnxruntime/onnxruntime/core/protobuf/onnx-ml.proto

**System information**
- ONNX Runtime version (you are using):

**Describe the solution you'd like**

I am trying to move build of protoc.exe to Unicode. After enabling UNICODE preprocessor and Unicode flag for the build it appears some issue is in promoting ascii strings to Unicode for the parameters passed to executable.

**Describe alternatives you've considered**
Currently converting argument parsing routines to consider Unicode characters.

**Additional context**
C:/чюфўшщ/ушЄ/microsoft/onnxruntime/onnxruntime/core/protobuf: warning: directory does not exist.
C:/чюфўшщ/ушЄ/microsoft/onnxruntime/onnxruntime/core/protobuf/onnx-ml.proto: No such file or directory

Please note garbled symbols чюфўшщ/ушЄ, these are characters corrupted due to incorrect handling of Unicode.

**Comment by snnn** [comment]
As onnxruntime is statically linked to protobuf, they should share the same preprocessor definitions. Right now, onnxruntime only support ANSI, not Unicode. Therefore you can't enable UNICODE for protobuf.

**Comment by yuslepukhin** [comment]
@MironAtHome Let's make it clear, you are attempting to run protoc on files that reside in paths that have UNICODE characters within them. Was protobuf code written in a way that it can handle UNICODE characters? Addining `_DUNICODE` switches to the command line is not enough.
`-DUNICODE` is a Microsoft specific switch which may affect lots of stuff , but then again, affect nothing depending on the code is written.
I took a brief look at protobuf code on GitHub and it looks like it is written in a way that it supports Unicode paths regardless of `-DUNICODE` at least here:
https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/io/io_win32.cc#L317
But it requires that the rest of the code treats path strings as UTF-8 encoding, otherwise it will attempt to convert to a code path CP_ACP in this case, this is almost never what you want.  You will notice that 
https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/io/io_win32.cc#L383
takes a flag `bool inUtf8` that denotes the string conversion from multibyte to widechar which is UTF-16 on Windows.
Bottom line, if you are able to build protobuf outside onnxruntime in  way that it handles UNICODE paths, we will make sure it does the same within onnxruntime.

**Comment by MironAtHome** [comment]
Just found your comments. Will review at home tonight what is new in the build ( if anything ) and where I have left the project off when worked on it.
I think I fixed it back than but need to take a look.

**Comment by faxu** [comment]
@mironathome is this still relevant and require fixing?
